src/main/java/dev/snowdrop/release/services/AutoUpdateService.java

```diff
@@ -0,0 +1,165 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class AutoUpdateService {
+    private static final Logger LOG = Logger.getLogger(AutoUpdateService.class);
+    private static final String GAV_CAMEL_CASE_REGEX_PATTERN = "[ -\\.]";
+    Matcher GAV_NAME_REGEX_PATTERN = Pattern.compile("([\\*\\+\\s]*)([0-9a-zA-Z\\.\\-]*)(:)([0-9a-zA-Z\\-]*)(\\s\\(.*[\\r\\n]?)").matcher("");
+    @Inject
+    BuildConfigFactory factory;
+
+    @Inject
+    IssueService issueSvc;
+
+    /**
+     * <p>Parses the comments from the yaml file to obtain variable definitions. These start with #!.</p>
+     *
+     * @param buildConfig <p>Contents of the Build Configuration file</p>
+     * @return
+     */
+    private static Map<String, String> readVariablesFromFile(InputStream buildConfig) {
+        String contents = "";
+        @SuppressWarnings("resource")
+        Scanner s = new Scanner(buildConfig);
+
+        s.useDelimiter("\\A");
+
+        if (s.hasNext()) {
+            contents = s.next();
+        }
+        LinkedHashMap<String, String> variables = new LinkedHashMap<>();
+        // Look for the variable defs i.e. #!myVariable=1.0.0
+        List<String> matches = getAllMatches(contents, "#![a-zA-Z0-9_-]*=.*");
+        for (String temp : matches) {
+            String[] nameValue = temp.split("=", 2);
+            // remove the !# of the name
+            variables.put(nameValue[0].substring(2), nameValue[1]);
+        }
+        return variables;
+    }
+
+    private static List<String> getAllMatches(String text, String regex) {
+        List<String> matches = new ArrayList<>();
+        Matcher m = Pattern.compile(regex).matcher(text);
+        while (m.find()) {
+            matches.add(m.group());
+        }
+        return matches;
+    }
+
+    public static String toCamelCase(final String init) {
+        if (init == null)
+            return null;
+        final StringBuilder ret = new StringBuilder(init.length());
+        for (final String word : init.split(GAV_CAMEL_CASE_REGEX_PATTERN)) {
+            if (!word.isEmpty()) {
+                ret.append(Character.toUpperCase(word.charAt(0)));
+                ret.append(word.substring(1).toLowerCase());
+            }
+        }
+        return ret.toString();
+    }
+
+    /**
+     * <p>Main process for updating the Build Configuration file.</p>
+     *
+     * @param repo
+     * @param release        <p>Release object</p>
+     * @param releaseVersion <p>Release version number</p>
+     * @param qualifier
+     * @param milestone
+     * @return
+     * @throws Throwable
+     */
+    public File updateBuildConfig(File repo, Release release, final String releaseVersion, final String qualifier, final String milestone) {
+        try {
+            final String[] releaseMMFA = releaseVersion.split("\\.");
```

Comment:
Please use a more explicit variable name.

Reply:
renamed it to `releaseVersionMajorMinorFix`.